### PR TITLE
added get thread member endpoint

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
## Checklist
- [x] I have tested this PR[^1].
- [x] I have read the [contributing guidelines](./CONTRIBUTING.md).

## Changelog
- Added the ability to get a thread member by their user id.
- Deprecated getThreadMembers in favor of requestThreadMembers

## Description

Closes: https://github.com/Javacord/Javacord/issues/906

[^1]: At least started a running bot instance with your changes and triggered an event so your changed code gets executed.

### CODE REVIEW

**Comment by RealYusufIsmail** [comment]
@Saladoc can you review this pr please

File: javacord-api/src/main/java/org/javacord/api/entity/channel/ThreadMember.java [hunk 1]
```diff
@@ -64,5 +64,4 @@ default CompletableFuture<User> requestUser() {
      * @return Any user-thread settings.
      */
     int getFlags();
-
```
Comment (Mysterypotatoguy) [inline javacord-api/src/main/java/org/javacord/api/entity/channel/ThreadMember.java:67]: Revert line removal

File: javacord-api/src/main/java/org/javacord/api/entity/channel/ServerThreadChannel.java [hunk 2]
```diff
@@ -182,6 +182,24 @@ default CompletableFuture<Void> leaveThread() {
         return getServer().leaveServerThreadChannel(getId());
     }
 
+    /**
+     * Used to get a thread member by their id.
+     *
+     * @param userId The id of the user.
+     * @return The thread member.
+     */
+    CompletableFuture<ThreadMember> getThreadMember(long userId);
```
Comment (Mysterypotatoguy) [inline javacord-api/src/main/java/org/javacord/api/entity/channel/ServerThreadChannel.java:191]:
```suggestion
    CompletableFuture<ThreadMember> requestThreadMemberById(long userId);
```

File: javacord-api/src/main/java/org/javacord/api/entity/channel/ServerThreadChannel.java [hunk 3]
```diff
@@ -182,6 +182,24 @@ default CompletableFuture<Void> leaveThread() {
...
+
+    /**
+     * Used to get a thread member by their id.
+     *
+     * @param userId The id of the user.
+     * @return The thread member.
+     */
+    default CompletableFuture<ThreadMember> getThreadMember(String userId) {
```
Comment (Mysterypotatoguy) [inline javacord-api/src/main/java/org/javacord/api/entity/channel/ServerThreadChannel.java:199]:
```suggestion
    default CompletableFuture<ThreadMember> requestThreadMemberById(String userId) {
```

**Review by Saladoc — CHANGES_REQUESTED** [review]
Methods that (might) do a rest request under the hood should be named `requestXY`. Methods that return a local value (for example from cache) should be named `getXY`.

Also please keep your method descriptions consistent. We usually describe what a method does (`Gets a Foo from the Bar`), not what the method is used for (`Used to get a Foo from the Bar`)

**Review by Mysterypotatoguy — CHANGES_REQUESTED** [review]
I'd argue `getThreadMembers` may need to be renamed to `requestThreadMembers` too, but this would be a breaking change

**Comment by Saladoc** [comment]
> I'd argue `getThreadMembers` may need to be renamed to `requestThreadMembers` too, but this would be a breaking change

We can rename the method, keep an old one in that calls the new one and deprecate it in favor of the new one.

File: javacord-api/src/main/java/org/javacord/api/entity/channel/ServerThreadChannel.java [hunk 4]
```diff
@@ -182,13 +182,31 @@ default CompletableFuture<Void> leaveThread() {
...
+    }
+
     /**
      * Gets all members in this thread.
      * Requires the {@link org.javacord.api.entity.intent.Intent#GUILD_MEMBERS} intent.
      *
      * @return All members in this thread.
      */
-    CompletableFuture<Set<ThreadMember>> getThreadMembers();
+    CompletableFuture<Set<ThreadMember>> requestThreadMembers();
```
Comment (Saladoc) [inline javacord-api/src/main/java/org/javacord/api/entity/channel/ServerThreadChannel.java:209]: This is a breaking change. Please restore the old method and deprecate it in favor of the new method.

**Review by Saladoc — CHANGES_REQUESTED** [review]

**Review by Saladoc — APPROVED** [review]

**Review by Mysterypotatoguy — APPROVED** [review]

**Pull request merged**